# Assignment 1

# Question 1

Use Python to generate random annual returns for 100 years for this asset. Compare the annual mean and volatility of returns from your random data to the true 10% mean and 20%.

In [3]:
import numpy as np

In [4]:
def random_annual_returns(mean_normal : float, volatility_normal: float, nb_years: float):
    returns_normal=np.random.normal(mean_normal,volatility_normal,nb_years) #100 values
    annual_mean = np.mean(returns_normal) #Calculate the mean
    annual_volatility = np.std(returns_normal) #Calculate the volatility
    return list((annual_mean, annual_volatility)) #I return the values in a list for a better visibility

In [9]:
np.random.seed(123)

mean_normal = 0.10
volatility_normal = 0.20
nb_years = 100
L_annual_values=random_annual_returns(mean_normal, volatility_normal, nb_years)
L_annual_values

[0.10542181469807196, 0.2256480940955922]

In [10]:
L_variation=[(L_annual_values[0]-mean_normal)/mean_normal*100,(L_annual_values[1]-volatility_normal)/volatility_normal*100] #Calculate the percentage of variation
L_variation

[5.421814698071959, 12.82404704779609]

# Conclusion :
- We can see that the mean is 5% higher
- We can see that the volatility is 13% higher

# Question 2

Now, generate random monthly returns for 100 years for this risky asset. Compare the annual mean and volatility of returns from your random data to the true 10% mean and 20%.

In [11]:
import numpy as np

In [19]:
def random_annual_monthly_returns(mean_normal : float, volatility_normal: float, nb_year : float,nb_months: float):
    mean_normal_monthly=mean_normal/nb_months #calculate the monthly mean
    volatility_normal=volatility_normal/np.sqrt(nb_months) #calculate the monthly volatility
    returns_normal=np.random.normal(mean_normal_monthly,volatility_normal,nb_months*nb_years) # 1200 Values 
    returns_normal_monthly_annual=returns_normal.reshape(-1,nb_months) # 100 lines and 12 columns , each line corresponds to a year
    returns_normal_annual=np.sum(returns_normal_monthly_annual,axis=1)  #sum the columns             
    annual_mean = np.mean(returns_normal_annual) #calculate the mean
    annual_volatility = np.std(returns_normal_annual) #calculate the volatility
    return list((annual_mean,annual_volatility))

In [20]:
np.random.seed(123)

mean_normal = 0.10
volatility_normal = 0.20
nb_year=100
nb_months = 12
L_monthly_annual_value=random_annual_monthly_returns(mean_normal, volatility_normal, nb_year, nb_months)
L_monthly_annual_value

[0.07886108435472816, 0.20358836159173122]

In [21]:
L_variation_monthly_annual=[(L_monthly_annual_value[0]-mean_normal)/mean_normal*100,(L_monthly_annual_value[1]-volatility_normal)/volatility_normal*100] #Calculate the percentage of variation
L_variation_monthly_annual

[-21.138915645271844, 1.794180795865602]

# Conclusion:
- We can see that the mean is 21% lower
- We can see that the volatility is 1.8% higher

# Question 3

Finally, generate random daily returns for 100 years for this risky asset. Compare the annual mean and volatility of returns from your random data to the true 10% mean and 20%.

In [22]:
import numpy as np

In [23]:
def random_annual_daily_returns(mean_normal : float, volatility_normal: float, nb_year, nb_days: float):
    mean_normal_daily=mean_normal/nb_days
    volatility_normal_daily=volatility_normal/np.sqrt(nb_days)
    returns_normal=np.random.normal(mean_normal_daily,volatility_normal_daily,nb_year*nb_days) # 25200 Values 
    returns_normal_daily_annual=returns_normal.reshape(-1,nb_days) # 100 lines and 252 columns, each line corresponds to a year
    returns_normal_annual=np.sum(returns_normal_daily_annual,axis=1)               
    annual_mean = np.mean(returns_normal_annual)
    annual_volatility = np.std(returns_normal_annual)
    return list((annual_mean,annual_volatility))

In [24]:
np.random.seed(123)
initial_price = 100
mean_normal = 0.10
volatility_normal = 0.20
nb_year=100
nb_days =252
L_daily_annual_value=random_annual_daily_returns(mean_normal, volatility_normal, nb_year, nb_days)
L_daily_annual_value

[0.1162987603731596, 0.2364135585285291]

In [25]:
L_variation_daily_annual=[(L_daily_annual_value[0]-mean_normal)/mean_normal*100,(L_daily_annual_value[1]-volatility_normal)/volatility_normal*100] #Calculate the percentage of variation
L_variation_daily_annual

[16.298760373159595, 18.206779264264547]

# Conclusion :
- We can see that the mean is 16% higher
- We can see that the volatility is 18% higher

# Question 4

What do you conclude from the results of the three previous questions?

From our analysis of the results of the three questions, I conclude that as we move from daily to monthly and then to annual returns, the estimated mean return approaches the true mean of 10% per year. And It is the same for volaitlity.